### DISCUSSION TITLE
[v3] $wire:init

### DISCUSSION THREAD

**Discussion by psjavi** [livewire/livewire#6885] [1. Help]
Hi, I'm migrating from v2 to v3, 

I've got a lot of components done in the following way:

`<div class="row grid-margin" wire:init="init">`

and I'm getting this error:

`
livewire.esm.js:1789 Alpine Expression Error: func(...).catch is not a function

Expression: "$wire.init"

 div.row.grid-margin
handleError @ livewire.esm.js:1789
tryCatch @ livewire.esm.js:1784
evaluate @ livewire.esm.js:1806
eval @ livewire.esm.js:8507
eval @ livewire.esm.js:6826
trigger @ livewire.esm.js:5938
eval @ livewire.esm.js:7689
eval @ livewire.esm.js:7688
eval @ livewire.esm.js:2436
eval @ livewire.esm.js:1470
eval @ livewire.esm.js:1470
walk @ livewire.esm.js:1393
eval @ livewire.esm.js:1468
deferHandlingDirectives @ livewire.esm.js:1947
initTree @ livewire.esm.js:1467
eval @ livewire.esm.js:1422
start2 @ livewire.esm.js:1421
start @ livewire.esm.js:7695
eval @ app.js:7
./resources/js/app.js @ app.js?id=3f0e756801c99fb5e2cd0b7c375a1fd3:646
__webpack_require__ @ app.js?id=3f0e756801c99fb5e2cd0b7c375a1fd3:808
(anónimo) @ app.js?id=3f0e756801c99fb5e2cd0b7c375a1fd3:967
__webpack_require__.O @ app.js?id=3f0e756801c99fb5e2cd0b7c375a1fd3:850
(anónimo) @ app.js?id=3f0e756801c99fb5e2cd0b7c375a1fd3:970
(anónimo) @ app.js?id=3f0e756801c99fb5e2cd0b7c375a1fd3:972
livewire.esm.js:1858 Uncaught TypeError: func(...).catch is not a function
    at eval (livewire.esm.js:1858:56)
    at tryCatch (livewire.esm.js:1782:24)
    at Object.evaluate (livewire.esm.js:1806:36)
    at eval (livewire.esm.js:8507:29)
    at Array.eval (livewire.esm.js:6826:7)
    at trigger (livewire.esm.js:5938:32)
    at eval (livewire.esm.js:7689:9)
    at Array.forEach (<anonymous>)
    at eval (livewire.esm.js:7688:18)
    at eval (livewire.esm.js:2436:68)
`

How can I solve it without a big refactor?

Thanks in advance

**Comment by lrljoe** [comment]
What are you trying to achieve there exactly?  

I'd suggest for one renaming your "init" function to something less likely to cause issues.

Next up - what's the content of your current "init" function?  Where are you loading it in from, in-line, or in your app.js?

**Comment by psjavi** [comment]
Ok, so I guess that now "init" it's an internal function, I'll try to change it.

Thanks.

**Reply by PhiloNL** [reply]
`wire:init` should still work. Please provide an example that is easy to replicate so we can verify if this is a bug. Thanks!

**Comment by psjavi** [comment]
In the component I've got:

`
    public function mount(): void
    {
       ...
    }
    public function init(MyService $myService): void
    {
        ...
    }
    public function render()
    {
        ...
        return view('livewire.my-view');
    }
`
    
In the partial I've got:

`<div wire:init="init"> ..... </div>`
